### ISSUE TITLE
persistence issue

### ISSUE THREAD

**Issue by ghost** [padenot/up-to-eleven#6]
when I set the state to "on" (dial to the right) it works
but in another tab, the dial is still "on" (dial to the right) but it doesn't work, so I have to twitch it off and on again for it to work.

please make it so that when it's on, it stays on regardless of the tab (or make that optional)

**Comment by ghost** [comment]
I tried @dregimbal fork: https://github.com/dregimbal/up-to-eleven
(downloaded all files, put them in a zip with .xpi extension, and installed it on firefox nightly with `xpinstall.signatures.required false`)
The button appears, and the settings do too, but it doesn't seem to do anything and I hear no difference enabling or disabling it or changing any of the settings.

**Comment by dregimbal** [comment]
@horusra can you verify you don't have any other extensions that might be interfering?

It's been a while since I've looked at my fork of this project, but I can confirm that it is still working with my install (Firefox 92.0.1). 
On my fork, I believe the link containing the \<video\> tag must start with "https://www.youtube.com/watch", for example "https://www.youtube.com/watch?v=23lRkdDXqY0". Could you try this link, or running the extension in a new [profile](https://support.mozilla.org/en-US/kb/profile-manager-create-remove-switch-firefox-profiles)?

As an alternative, your sound card may have a "Loudness Equalization" or similar setting which may help depending on your use-case

**Comment by padenot** [comment]
Yeah so the problem is that the button is global, but only affects the tab. @dregimbal does something a lot better it seems like, I had a quick look.

@dregimbal, do you want to merge all your commit here and, I can publish a new version, and give you commit permission, and then we can publish a new version? We can extend the logic that restricts to some sites, since you helpfully added an option panel. That sounds useful.

**Comment by horusra** [comment]
> 
> 
> @horusra can you verify you don't have any other extensions that might be interfering?
> 
On my fork, I believe the link containing the <video> tag must start with "https://www.youtube.com/watch", for example "https://www.youtube.com/watch?v=23lRkdDXqY0".

Oh, that must have been the issue, I didn't know it only worked on youtube, I was using a different video streaming site when I tested it.

**Comment by ghost** [comment]
> 
> 
> @horusra can you verify you don't have any other extensions that might be interfering?
> 
> It's been a while since I've looked at my fork of this project, but I can confirm that it is still working with my install (Firefox 92.0.1). On my fork, I believe the link containing the video tag must start with "https://www.youtube.com/watch", for example "https://www.youtube.com/watch?v=23lRkdDXqY0". 

Oh, that must have been the issue, I didn't know it only worked on youtube, I was using a different video streaming site when I tested it.

Edit: Installed your fork again just now, and tested it on youtube, works well! However I would for it to not be specific to youtube or just certain websites.

>As an alternative, your sound card may have a "Loudness Equalization" or similar setting which may help depending on your use-case

Yes, it does, but that means I lose all that when I switch to my bluetooth headphones or some other source. The other issue is also when you don't want it to affect all sound sources, like when you're playing music from a music player and watching or listening to web sources from time to time, like youtube or some online lecture. So a browser specific solution is great for that (since audio quality has lower priority than consistent and loud enough volume)
